src/sampleutil.cpp

```diff
@@ -178,8 +178,14 @@ void SampleUtil::copyWithRampingGain(CSAMPLE* pDest, const CSAMPLE* pSrc,
 // static
 void SampleUtil::convertS16ToFloat32(CSAMPLE* pDest, const SAMPLE* pSrc,
         unsigned int iNumSamples) {
+    // Sample min can be -32768 instead of -32767, so make sure our conversion
+    // scale is picking the largest possible absolute value.
+    // Note that this means that although some sample values convert to -1.0,
+    // none will convert to +1.0.
+    static const CSAMPLE kConversionFactor = math_max(CSAMPLE(-SAMPLE_MIN),
```

Comment:
Since SHRT_MIN seams to be unreliable, do not use it you also do not know what SHRT_MAX is in the SHRT_MIN = -32767 case. Just use 0x8000 (it is very self explaining as hex constant) this is without a doubt the greater or equal  SHRT_MAX and works on all known architectures. 
If not portaudio and libmad will also not work.